src/query/processor/cypher/semanticanalyzer/SemanticAnalyzer.cpp

```diff
@@ -0,0 +1,261 @@
+/**
+Copyright 2024 JasmineGraph Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+#include <iostream>
+#include "SemanticAnalyzer.h"
+#include "ScopeManager.h"
+
+
+using namespace std;
+// Constructor
+SemanticAnalyzer::SemanticAnalyzer() {
+    // Initialize the scope manager with the global scope
+    scopeManager = new ScopeManager();
+    temp = new unordered_map<string, string>();// Enter global scope
+}
+
+// Destructor
+SemanticAnalyzer::~SemanticAnalyzer() {
+    // Ensure to exit all scopes upon destruction
+    scopeManager->exitScope();
+}
+
+
+// Main method to analyze the AST
+bool SemanticAnalyzer::analyze(ASTNode* root, bool canDefine, string type) {
+    if(root->nodeType == "VARIABLE" && canDefine)
+    {
+        if(!checkVariableDeclarations(root, type))
+        {
+            return false;
+        }
+    }else if(root->nodeType == "VARIABLE")
```

Comment:
Can we move these literals to some constants and refer them here? I see same literal repeated several times here.

Reply:
sure sir, I will do.

Reply:
![image](https://github.com/user-attachments/assets/d9fa29a4-e6fe-41e3-8e59-7b472ce9efce)

sir, is it okay to move the constants to the const.cpp file in the util folder ?

Reply:
Yes, Its ok the way you have set the util folder currently.